outputs/jira_test.go

```diff
@@ -0,0 +1,365 @@
+			priorities: nil,
+			wantError:  "project don't have issue priorities",
+		},
+		{
+			name:       "bad path (jira returns error)",
+			jiraAPI:    &JiraAPI{Priority: ""},
+			priorities: jira.Issue{},
+			wantError:  "failed to get issue priority list: json: cannot unmarshal object into Go value of type []jira.Priority",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				propertiesJson, err := json.Marshal(test.priorities)
+				if err != nil {
+					t.Errorf("Unexpected json mashal error: %v", err)
+				}
+				_, err = w.Write(propertiesJson)
+				if err != nil {
+					t.Errorf("Unexpected write response error: %v", err)
+				}
+			}))
+			defer ts.Close()
+
+			jiraClient, err := jira.NewClient(ts.Client(), ts.URL)
+			if err != nil {
+				t.Fatalf("can't create jiraClient %v", err)
+			}
+
+			err = createIssuePriority(test.jiraAPI, jiraClient)
+
+			if test.wantError != "" {
+				require.NotNil(t, err)
+				assert.Contains(t, err.Error(), test.wantError)
+			} else {
+				require.Equal(t, test.wantPriority, test.jiraAPI.Priority)
+			}
+		})
+	}
+}
+
+func TestCreateIssueType(t *testing.T) {
+	tests := []struct {
+		name          string
+		jiraAPI       *JiraAPI
+		metaProject   *jira.MetaProject
+		wantIssueType string
+		wantError     string
+	}{
+		{
+			name:          "happy path (empty issueType, jira have 'Task' field)",
+			jiraAPI:       &JiraAPI{},
+			metaProject:   &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{{Name: "Task"}, {Name: "Bug"}}},
+			wantIssueType: "Task",
+		},
+		{
+			name:          "happy path (empty issueType, jira doesn't have 'Task' field)",
+			jiraAPI:       &JiraAPI{},
+			metaProject:   &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{{Name: "Story"}, {Name: "Bug"}}},
+			wantIssueType: "Story",
+		},
+		{
+			name:          "happy path (fill priority, jira have 'Bug' field)",
+			jiraAPI:       &JiraAPI{Issuetype: "Bug"},
+			metaProject:   &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{{Name: "Task"}, {Name: "Bug"}}},
+			wantIssueType: "Bug",
+		},
+		{
+			name:        "bad path (fill priority, jira doesn't have 'Bug' field)",
+			jiraAPI:     &JiraAPI{Issuetype: "Bug"},
+			metaProject: &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{{Name: "Task"}, {Name: "Story"}}},
+			wantError:   "project \"\" don't have issueType \"Bug\"",
+		},
+		{
+			name:        "bad path (metaIssueType have empty IssueTypes)",
+			jiraAPI:     &JiraAPI{Priority: ""},
+			metaProject: &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{}},
+			wantError:   "project \"\" don't have issueTypes",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := createIssueType(test.jiraAPI, test.metaProject)
+
+			if test.wantError != "" {
+				require.NotNil(t, err)
+				assert.Contains(t, err.Error(), test.wantError)
```

Comment:
```suggestion
				require.NotNil(t, err, test.name)
				assert.Contains(t, err.Error(), test.wantError, test.name)
```

Reply:
i don't think that we need these changes
we already have `test.name` in the `Test` field:
```
Error:          Expected value not to be nil.
Test:           TestCreateIssueType/happy_path_(empty_issueType,_jira_have_'Task'_field)
Messages:       happy path (empty issueType, jira have 'Task' field)     // duplication
```

Reply:
ok fair enough